### PR TITLE
Dockerfile to assemble a deployable image

### PR DESCRIPTION
**Pull request by sbose78**
Build the image in a single Dockerfile - helps to build an image in a containerized environment with tools like BuildConfig and Tekton.

### CODE REVIEW

**Comment by sbose78** [comment]
cc @EricWittmann @pmuir

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.4-210 as builder
```
Comment (pmuir) [inline Dockerfile:1]: do we want to use this or quay?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+RUN export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-11.0.13.0.8-3.el8_5.x86_64 \
+    && export M2_HOME=/opt/maven \
+    && export MAVEN_HOME=/opt/maven \
+    && export PATH=$M2_HOME/bin:$PATH \
+    && make build-project 
+
+WORKDIR /release/
+RUN tar -xf /workspace/distro/docker/target/docker/app-files/apicurio-registry-storage-sql-$VERSION-all.tar.gz -C /release
+
```
Comment (pmuir) [inline Dockerfile:23]: I like to put a comment in here to provide a visual clue to the multi-part docker file

File: Dockerfile (hunk 1 above)
Comment (EricWittmann) [reply Dockerfile:1]: Since this is ultimately a downstream dockerfile, I could see it either way.

File: Dockerfile (hunk 2 above)
Comment (EricWittmann) [reply Dockerfile:23]: +1

**Review by EricWittmann — APPROVED** [review]

**Comment by famartinrh** [comment]
btw notice we build two images for two components in this repository. apicurio-registry and tenant-manager. The dockerfile added here is only for apicurio-registry component.

**Comment by EricWittmann** [comment]
Very good point.  The tenant manager is a sidecar and should also be built.  Not sure if it should be built in this same Dockerfile, but maybe??

**Comment by famartinrh** [comment]
probably not in the same dockerfile, I think we just need two dockerfiles

**Pull request merged**